src/chunk_constraint.h

```diff
@@ -11,9 +11,20 @@
 #include "ts_catalog/catalog.h"
 #include "hypertable.h"
 
+typedef enum CConstrType /* types of chunk constraints */
+{
+	CCONSTR_NONE,
+	CCONSTR_DIMENSION,
+	CCONSTR_CHECK,
+	CCONSTR_CORRELATED
+} CConstrType;
```

Comment:
Can we use the DimensionType enum here too? 

DIMENSION would be an alias for `DIMENSION_TYPE_OPEN || DIMENSION_TYPE_CLOSED`
CHECK would be DIMENSION_TYPE_ANY. It would save us an extra enum and additional translation.

Reply:
in the `DIMENSION` speak, `ANY` actualy means `NONE` in some ways and we don't represent `CHECK` constraints in the dimension world (although they are basically a PG representation for the OPEN/CLOSED dimensions. I'd rather have these few enums here to keep things explicit.

Reply:
Done